exercises/meetup/src/example.c

```diff
@@ -0,0 +1,123 @@
+#include <string.h>
+#include "meetup.h"
+
+#define BAD_DATE_REQUESTED (0)
+
+typedef struct {
+   const char *text;
+   int number;
+} textToIntLookup_t;
+
+static int daysInMonth(int year, int month)
+{
+   int numberOfDays = 31;
+
+   switch (month) {
+   case 4:
+   case 6:
+   case 9:
+   case 11:
+      numberOfDays = 30;
+      break;
+   case 2:
+      if ((((year % 4) == 0) && ((year % 100) != 0)) || ((year % 400) == 0)) {
+         numberOfDays = 29;
+      } else {
+         numberOfDays = 28;
+      }
+      break;
+   }
+   return numberOfDays;
+}
+
+static int getNumberGivenText(const char *text, textToIntLookup_t * lookup,
+                              size_t numberEntries)
+{
+   int matchNumber = -1;
+
+   for (size_t index = 0; index < numberEntries; index++) {
+      if (0 == strcmp(text, lookup[index].text)) {
+         matchNumber = lookup[index].number;
+         break;
+      }
+   }
+   return matchNumber;
+}
+
+static int getDayOfWeek(unsigned int year, unsigned int month, unsigned int day)
+{
+   struct tm firstDayOfTestMonth;
+   time_t midnightFirstDayofTestMonth;
+
+   firstDayOfTestMonth.tm_year = year - 1900;
+   firstDayOfTestMonth.tm_mon = month - 1;
+   firstDayOfTestMonth.tm_mday = day;
+   firstDayOfTestMonth.tm_hour = 0;
+   firstDayOfTestMonth.tm_min = 0;
+   firstDayOfTestMonth.tm_sec = 0;
+   firstDayOfTestMonth.tm_isdst = 0;
+   midnightFirstDayofTestMonth = mktime(&firstDayOfTestMonth);
+
+   return (localtime(&midnightFirstDayofTestMonth)->tm_wday);
+}
+
+int meetupDayOfMonth(unsigned int year, unsigned int month, const char *week,
+                     const char *dayOfWeek)
+{
+   int dayOfMonth = BAD_DATE_REQUESTED;
+
+   textToIntLookup_t dayLookup[] = {
+      {"Sunday", 0},
+      {"Monday", 1},
+      {"Tuesday", 2},
+      {"Wednesday", 3},
+      {"Thursday", 4},
+      {"Friday", 5},
+      {"Saturday", 6}
+   };
+
+   textToIntLookup_t whichWeekOfMonthLookup[] = {
+      {"first", 1},
+      {"second", 2},
+      {"third", 3},
+      {"fourth", 4},
+      {"fifth", 5},
+      {"last", 6},
+      {"teenth", 7}
+   };
+
+   int baselineDayOfWeek;
+   int daysInTestMonth = daysInMonth(year, month);
+   int dayOffset;
+   int referenceDay = 1;        // used to calculate a baseline day of week given a target date.
+
+   int whichWeekOfMonth = getNumberGivenText(week, &whichWeekOfMonthLookup[0],
```

Comment:
Since this is the only use of`whichWeekOfMonthLookup`, you might consider refactoring it into a helper `getWeekOfMonth()`.